src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -1994,7 +1994,8 @@ public Object poolKey(Types types) {
 
         @Override @DefinedBy(Api.LANGUAGE_MODEL)
         public Set<Modifier> getModifiers() {
-            long flags = flags();
+            // just in case the method is restricted but that is not a modifier
+            long flags = flags() & ~RESTRICTED;
             return Flags.asModifierSet((flags & DEFAULT) != 0 ? flags & ~ABSTRACT : flags);
```

Comment:
Maybe `Flags::asModifierSet` could be extended in another PR to take a parameter for the different modifier kinds, which would then be used to disambiguate overloaded `Flags` bits.

Reply:
yep we could have other instances of this same issue in the future